cpp/unit/ability.cpp

```diff
@@ -1,88 +1,242 @@
 // Copyright 2014-2015 the openage authors. See copying.md for legal info.
 
-#include "ability.h"
-
 #include <memory>
 
 #include "../terrain/terrain_object.h"
+#include "../datamanager.h"
+#include "ability.h"
 #include "action.h"
+#include "command.h"
 #include "unit.h"
 
 namespace openage {
 
-MoveAbility::MoveAbility(Texture *t, TestSound *s)
+std::string to_string(const ability_type &at) {
```

Comment:
That's not the way I would do it. I'd rather do something like :
```
namespace  openage {
    using std::to_string;
    std::string to_string(const ability_type &at);
 }

// use openage::to_string everywhere 
```
I always prefer not to specialize in std if I can avoid it .  Plus in this particular case it is overloading not specialization so it is forbidden by the way :)

http://stackoverflow.com/questions/17136497/is-overriding-stdto-string-for-user-defined-enums-proper-way-to-privide-to-str